### DISCUSSION TITLE
How to know if e.g. pagination is enabled (from existing table instance)?

### DISCUSSION THREAD

**Discussion by marceloverdijk** [TanStack/table#5306] [Q&A]
If I have a `table` instance, how can I check if that table has pagination enabled?
Same I would like to know if a global filter was configured when the table was created.

I was thinking something like:

```
const paginationEnabled = table.getState().pagination !== undefined
```

but even for a simple table created like:

```
  const table = useReactTable({
    data,
    columns,
    getCoreRowModel: getCoreRowModel(),
  })
```

the `state` returns:

```
{
  "columnSizing": {},
  "columnSizingInfo": {
    "startOffset": null,
    "startSize": null,
    "deltaOffset": null,
    "deltaPercentage": null,
    "isResizingColumn": false,
    "columnSizingStart": []
  },
  "rowSelection": {},
  "expanded": {},
  "grouping": [],
  "sorting": [],
  "columnFilters": [],
  "columnPinning": {
    "left": [],
    "right": []
  },
  "rowPinning": {
    "top": [],
    "bottom": []
  },
  "columnOrder": [],
  "columnVisibility": {},
  "pagination": {
    "pageIndex": 0,
    "pageSize": 10
  }
}
```

The table itself still displays all rows (even if there are more then 10) so internally Tanstack Table looks at something different then the `pagination` state to determine if pagination is enabled.

See https://codesandbox.io/p/sandbox/laughing-pine-x5qqvp

**Comment by goatfryed** [comment]
Hey @marceloverdijk,
I noticed that I asked the same question as you a couple of hours ago here #5342. My question is a bit more general as it's not just pagination specific, but yours is more detailed about the current struggles. Do you mind updating your question to be a bit more general, so it's easier to find for others? If you do, I'll close my question.

**Comment by marceloverdijk** [comment]
I'm now using:

```
const isPaginationEnabled = table.options.getPaginationRowModel !== undefined
```

and that gives `false` if no `getPaginationRowModel: getPaginationRowModel(),` is provided when creating the table, and `true` otherwise.
